### PR TITLE
Added in truncate and ftruncate system calls

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 1]
```diff
@@ -5548,6 +5548,40 @@ int32_t NaClSysAccess(struct NaClAppThread *natp,
   return ret;
 }
 
+int32_t NaClSysTruncate(struct NaClAppThread *natp,
+                        uint32_t file, int length) {
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_syscall_common.c:5552]: this is the path right? other calls just use char*, shouldnt that work?

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 2]
```diff
@@ -5548,6 +5548,40 @@ int32_t NaClSysAccess(struct NaClAppThread *natp,
...
+
+  ret = lind_truncate(path, length, nap->cage_id);
+
+  return ret;
+}
+
+int32_t NaClSysFtruncate(struct NaClAppThread *natp,
+                         int fd, int length) {
+  int32_t ret;
+  struct NaClApp *nap = natp->nap;
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_syscall_common.c:5575]: You never translate the fd from userfd to lind fd, which should be wrong. May have worked in some corner cases?

**Review by rennergade — CHANGES_REQUESTED** [review]

File: src/trusted/service_runtime/nacl_syscall_common.c (hunk 1 above)
Comment (jesings) [reply src/trusted/service_runtime/nacl_syscall_common.c:5552]: There's a comment in syscall_handlers_gen.py that says you should use a uint32_t but we've mostly been ignoring that. I think it makes more sense to follow that because we're really only passing 32 bits of stuff into NaCl here.

File: src/trusted/service_runtime/nacl_syscall_common.c (hunk 2 above)
Comment (jesings) [reply src/trusted/service_runtime/nacl_syscall_common.c:5575]: Yeah I think postgres was using truncate and it worked in the corner case of my test... I'll fix it

**Review by justyoyo123 — APPROVED** [review]
This pr that adds truncate and ftruncate into NaCl looks good to me.
